tensorboard/plugins/hparams/backend_context_test.py

```diff
@@ -358,6 +358,55 @@ def test_experiment_without_experiment_tag_many_distinct_values(self):
         _canonicalize_experiment(actual_exp)
         self.assertProtoEquals(expected_exp, actual_exp)
 
+    def test_experiment_with_bool_types(self):
+        self.session_1_start_info_ = """
+            hparams:[
+              {key: 'batch_size' value: {bool_value: true}}
+            ]
+        """
+        self.session_2_start_info_ = """
+            hparams:[
+              {key: 'batch_size' value: {bool_value: false}}
```

Comment:
What's interesting about the logic you've added is that the domain is [True, False] no matter what the values are in *start_info_. I think it would be worth acknowledging that by changing this test case (or adding new ones) so that only a single value (one of True or one of False) is specied in  *start_info_.

Reply:
That's a good point, I'll make them both `True` here so that it is more self evident